### DISCUSSION TITLE
How do I load a custom Whisper model (from HuggingFace)?

### DISCUSSION THREAD

**Discussion by athu16** [openai/whisper#1170] [Q&A] [answered]
I want to load [this fine-tuned model](https://huggingface.co/DrishtiSharma/whisper-large-v2-marathi) using my existing Whisper installation. I have a Python script which uses the `whisper.load_model()` function, but it only accepts strings like "small", "base", etc. How can I load a custom model for inference?

**Comment by jongwook** [comment, marked as answer]
The `load_model()` function also checks if the `name` string is a path to an existing file:

https://github.com/openai/whisper/blob/c09a7ae299c4c34c5839a76380ae407e7d785914/whisper/__init__.py#L130-L139

So you can simply supply the checkpoint path. It should still have the same format, containing a dictionary with `"dims"` and `"model_state_dict"` fields.

**Reply by anonymouscnuser** [reply]
This only load a .pt file. But in huggingface, most libraries don't provide checkpoints, but `pytorch_model.bin` files. And whisper library can't load them easily. Could you please offer a example that load a model from huggingface like https://huggingface.co/openai/whisper-medium or https://huggingface.co/vumichien/whisper-medium-jp ?

**Reply by phineas-pta** [reply]
you can inspire from this: https://github.com/openai/whisper/discussions/830#discussioncomment-4652413
